private/bufpkg/bufplugin/bufpluginref/bufpluginref.go

```diff
@@ -93,6 +93,21 @@ func PluginReferenceForString(reference string, revision int) (PluginReference,
 	return parsePluginReference(reference, revision)
 }
 
+// PluginReferenceOptionalVersion returns a new PluginReference for the given string.
+// If the string does not contain a version, the version is assumed to be an empty string.
+// This parses the path in the form remote/owner/plugin:version.
+func PluginReferenceOptionalVersion(rawReference string) (PluginReference, error) {
+	if reference, err := PluginReferenceForString(rawReference, 0); err == nil {
+		return reference, nil
+	}
+	// Try parsing as a plugin identity (no version information)
+	identity, err := PluginIdentityForString(rawReference)
+	if err != nil {
+		return nil, fmt.Errorf("invalid remote plugin %s", rawReference)
+	}
+	return &pluginReference{identity: identity}, nil
```

Comment:
I'm going to make the function return `(remote, owner, plugin, version string, err error)` because otherwise I need to replicate the logic everywhere of checking if the passed string is a valid reference, and if not then checking if it's a valid identity, and then setting `remote, owner, plugin, version` vars appropriately

Reply:
Also yeah I don't think it's a great idea to make a new struct `ReferenceOrIdentity`

Reply:
We do not want a shared function here that has 5 return values - there's other solutions, such as `func PluginIdentityAndVersionForString(string) (PluginIdentity, string, error)` that is documented to return empty for the version if no version is present.